analyzers/tests/SonarAnalyzer.UnitTest/TestCases/InsteadOfAny.EntityFramework.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Microsoft.EntityFrameworkCore;
+
+// https://github.com/SonarSource/sonar-dotnet/issues/7286
+public class EntityFrameworkReproGH7286
+{
+    public class MyEntity
+    {
+        public int Id { get; set; }
+    }
+
+    public class MyDbContext : DbContext
+    {
+        public DbSet<MyEntity> MyEntities { get; set; }
+    }
+
+    public List<MyEntity> GetEntities(MyDbContext dbContext, List<int> ids) =>
+        dbContext
+            .MyEntities
+            .Where(e => ids.Any(i => e.Id == i)) // Noncompliant - FP, should not raise in context of EntityFramework queries. Exist cannot be translated to SQL query
```

Comment:
We should add also the FP for S6617 (with `e.Id > i`).
Moreover, I would add also tests for `a.Equals(b)` and with pattern matching `a is b` for both.